src/libpmemfile-core/inode.h

```diff
@@ -79,6 +79,29 @@ struct pmemfile_vinode {
 	} snapshot;
 };
 
+struct inode_perms {
+	uint64_t flags;
+	uint32_t uid;
+	uint32_t gid;
+};
+
+static inline void
+_vinode_get_perms(struct pmemfile_vinode *vinode, struct inode_perms *perms)
```

Comment:
This might look nicer if it just returned a 'struct inode_perms'

static inline struct inode_perms _vinode_get_perms(...)

The optimizer probably generates exactly the same code either way.